Update header locations for uapi & generated

In the v3.6 kernel, many of the headers were split out into a uapi variant.
klibc should be using those variants to compile.

There is also a generated subdir, to seperate out generated headers.

The old kernel header location needs to be included as well, because
linux/compiler.h is available only in the base include directory.

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar klibc-2.0.3.orig/scripts/Kbuild.klibc klibc-2.0.3/scripts/Kbuild.klibc
--- klibc-2.0.3.orig/scripts/Kbuild.klibc	2013-12-03 10:53:46.000000000 -0800
+++ klibc-2.0.3/scripts/Kbuild.klibc	2013-12-26 14:39:07.113906101 -0800
@@ -77,7 +77,10 @@
 KLIBCCFLAGS       :=
 
 # Defaults for arch to override
-KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include
 
 # Arch specific definitions for klibc
 include $(srctree)/$(KLIBCSRC)/arch/$(KLIBCARCHDIR)/MCONFIG
@@ -116,6 +119,9 @@
-KLIBCCPPFLAGS    += -I$(KLIBCKERNELSRC)/include				    \
+KLIBCCPPFLAGS    += -I$(KLIBCKERNELSRC)/include/uapi				    \
+                    -I$(KLIBCKERNELSRC)/include/generated/uapi				    \
+                    -I$(KLIBCKERNELSRC)/include/generated				    \
+                    -I$(KLIBCKERNELSRC)/include				    \
                      $(if $(KBUILD_SRC),-I$(KLIBCKERNELOBJ)/include2 	    \
 		       -I$(KLIBCKERNELOBJ)/include -I$(srctree)/include)    \
 		     $(KLIBCARCHINCFLAGS)
 
 # klibc definitions
diff -Nuar klibc-2.0.3.orig/usr/klibc/arch/i386/MCONFIG klibc-2.0.3/usr/klibc/arch/i386/MCONFIG
--- klibc-2.0.3.orig/usr/klibc/arch/i386/MCONFIG	2013-12-03 10:53:46.000000000 -0800
+++ klibc-2.0.3/usr/klibc/arch/i386/MCONFIG	2013-12-26 14:44:56.349161344 -0800
@@ -34,3 +34,6 @@
 
 # Asm includes for i386 are in the merged x86 tree
-KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include
diff -Nuar klibc-2.0.3.orig/usr/klibc/arch/ppc/MCONFIG klibc-2.0.3/usr/klibc/arch/ppc/MCONFIG
--- klibc-2.0.3.orig/usr/klibc/arch/ppc/MCONFIG	2013-12-03 10:53:46.000000000 -0800
+++ klibc-2.0.3/usr/klibc/arch/ppc/MCONFIG	2013-12-26 14:44:56.350161362 -0800
@@ -24,6 +24,9 @@
 
 # The kernel so far has both asm-ppc* and asm-powerpc.
-KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include
 
 # The asm include files live in asm-powerpc
 KLIBCASMARCH	= powerpc
diff -Nuar klibc-2.0.3.orig/usr/klibc/arch/ppc64/MCONFIG klibc-2.0.3/usr/klibc/arch/ppc64/MCONFIG
--- klibc-2.0.3.orig/usr/klibc/arch/ppc64/MCONFIG	2013-12-03 10:53:46.000000000 -0800
+++ klibc-2.0.3/usr/klibc/arch/ppc64/MCONFIG	2013-12-26 14:44:56.350161362 -0800
@@ -22,6 +22,9 @@
 
 # The kernel so far has both asm-ppc* and asm-powerpc.
-KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include
 
 # The asm include files live in asm-powerpc
 KLIBCASMARCH	= powerpc
diff -Nuar klibc-2.0.3.orig/usr/klibc/arch/x86_64/MCONFIG klibc-2.0.3/usr/klibc/arch/x86_64/MCONFIG
--- klibc-2.0.3.orig/usr/klibc/arch/x86_64/MCONFIG	2013-12-03 10:53:46.000000000 -0800
+++ klibc-2.0.3/usr/klibc/arch/x86_64/MCONFIG	2013-12-26 14:44:54.522128579 -0800
@@ -37,3 +37,6 @@
 
 # Asm includes for x86_64 are in the merged x86 tree
-KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include
